File: app/src/main/java/com/zacharee1/rctdremoverforlg/misc/SwitchViewWithText.java
package com.zacharee1.rctdremoverforlg.misc;

import android.content.Context;
import android.content.res.TypedArray;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;
import android.view.ViewGroup;
import android.widget.CompoundButton;
import android.widget.LinearLayout;
import android.widget.Switch;
import android.widget.TextView;

import com.zacharee1.rctdremoverforlg.R;


public class SwitchViewWithText extends LinearLayout {
    private String subText;
    private String text;

    private Switch mSwitch;
    private TextView mSubText;
    private TextView mTitleText;

    public SwitchViewWithText(Context context, AttributeSet attrs) {
        super(context, attrs);

        inflate(getContext(), R.layout.switch_with_text, this);

        TypedArray a = context.getTheme().obtainStyledAttributes(
                attrs,
                R.styleable.SwitchViewWithText,
                0, 0);

        try {
            subText = a.getString(R.styleable.SwitchViewWithText_subText);
            text = a.getString(R.styleable.SwitchViewWithText_text);
        } finally {
            a.recycle();
        }

        mSwitch = findViewById(R.id.switchView);
        mTitleText = findViewById(R.id.titleText);
        mSubText = findViewById(R.id.subText);

        mTitleText.setText(text);
        mSubText.setText(subText);

        mSwitch.setFocusable(false);
        mTitleText.setFocusable(false);
        mSubText.setFocusable(false);

        if (subText == null || subText.isEmpty()) {
            mSubText.setVisibility(GONE);
        }
    }

    public void setChecked(boolean checked) {
        mSwitch.setChecked(checked);
    }

    public void setText(CharSequence text) {
        mTitleText.setText(text);
    }

    public void setSubText(CharSequence subText) {
        mSubText.setText(subText);
    }

    public void setOnCheckedChangeListener(CompoundButton.OnCheckedChangeListener listener) {
        mSwitch.setOnCheckedChangeListener(listener);
    }

    public boolean isChecked() {
        return mSwitch.isChecked();
    }

    public CharSequence getText() {
        return mTitleText.getText();
    }

    public CharSequence getSubText() {
        return mSubText.getText();
    }
}

File: app/src/main/java/com/zacharee1/rctdremoverforlg/misc/SwitchViewWithText.kt
package com.zacharee1.rctdremoverforlg.misc

import android.content.Context
import android.util.AttributeSet
import android.widget.CompoundButton
import android.widget.LinearLayout
import android.widget.Switch
import android.widget.TextView
import com.zacharee1.rctdremoverforlg.R

class SwitchViewWithText(context: Context, attrs: AttributeSet) : LinearLayout(context, attrs) {
    private val mSwitch: Switch?
    private val mSubText: TextView?
    private val mTitleText: TextView?

    private var subText: String?
        get() = mSubText?.text.toString()
        set(value) {
            mSubText?.text = value
        }

    private var text: String?
        get() = mTitleText?.text.toString()
        set(value) {
            mTitleText?.text = value
        }

    var isChecked: Boolean
        get() = mSwitch?.isChecked as Boolean
        set(value) {
            mSwitch?.isChecked = value
        }

    init {
        inflate(getContext(), R.layout.switch_with_text, this)

        val a = context.theme.obtainStyledAttributes(
                attrs,
                R.styleable.SwitchViewWithText,
                0, 0)

        mSwitch = findViewById(R.id.switchView)
        mTitleText = findViewById(R.id.titleText)
        mSubText = findViewById(R.id.subText)

        try {
            subText = a.getString(R.styleable.SwitchViewWithText_subText)
            text = a.getString(R.styleable.SwitchViewWithText_text)
        } finally {
            a.recycle()
        }

        mSwitch.isFocusable = false
        mTitleText.isFocusable = false
        mSubText.isFocusable = false

        if (subText == null || subText?.isEmpty() as Boolean) {
            mSubText.visibility = GONE
        }
    }

    fun setOnCheckedChangeListener(listener: (CompoundButton, Boolean) -> Unit) {
        mSwitch?.setOnCheckedChangeListener(listener)
    }
}